src/zm_monitor.cpp

```diff
@@ -557,7 +560,10 @@ bool Monitor::connect() {
   Debug(3, "Connecting to monitor.  Purpose is %d", purpose );
 #if ZM_MEM_MAPPED
   snprintf(mem_file, sizeof(mem_file), "%s/zm.mmap.%d", staticConfig.PATH_MAP.c_str(), id);
-  map_fd = open(mem_file, O_RDWR|O_CREAT, (mode_t)0600);
+  /* mode 660 to allow group www-data */
+  mode_t oldmask = umask(0002);
+  map_fd = open(mem_file, O_RDWR|O_CREAT, (mode_t)0660);
```

Comment:
I did a test here.  the umask change was not required.  On my machine the 0660 took effect.

Reply:
Here its mandatory. Global `umask` might come from the init system. Alpine has a custom init system and OpenRC.
```
$ ll /dev/shm/
total 502208
drwxrwxrwt    2 root     root            60 Dec  2 03:12 .
drwxr-xr-x    4 root     root           440 Dec  1 01:39 ..
-rw-r-----    1 zonemind www-data 514259144 Dec  2 03:12 zm.mmap.1
$ umask
0022

```

Reply:
Found a way in OpenRC to set `process umask` (start-stop-daemon-args) thus we are on par now. Only 660 needed.